pylint/checkers/utils.py

```diff
@@ -203,6 +203,48 @@ def is_builtin(name):
     """
     return name in builtins or name in SPECIAL_BUILTINS
 
+
+def is_defined_in_scope(var_node, varname, scope):
+    if isinstance(scope, astroid.If):
+        for node in scope.body:
+            if ((isinstance(node, astroid.AssignName) and node.name == varname)
```

Comment:
I don't think this actually does anything, since you cannot have standalone `AssignName` nodes. Most of the time `AssignName` is wrapped in an `Assign`, so you might want to look into that instead.